overlord/snapstate/handlers.go

```diff
@@ -470,7 +470,7 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 		if readInfoErr == nil {
 			break
 		}
-		if _, ok := readInfoErr.(*snap.NotFoundError); !ok {
+		if _, ok := readInfoErr.(*snap.BrokenSnapError); !ok {
```

Comment:
use the general error here seems strange. We are really checking for mounted vs not-mounted here.

Reply:
that's a bit strange,  it might be easier to keep NotFoundError as is, and use an interface to subsume both:

type BrokenSnapError interface {
   error
  Broken() string
}

Reply:
Ah, nice idea. I'll do that.